Spring Boot support breaks when Kotlin is being used

### Expected Behavior
Able to build docker image

### Current Behavior
Exit with an error

### Context
<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context helps us come up with a solution that is most useful in the real world -->
Broke all Spring Boot projects that use both Kotlin and Spring Boot support. Reproducible with an empty project.

### Steps to Reproduce (for bugs)
<!--- Provide a self-contained example project (as an attached archive or a Github project). -->
<!--- In the rare cases where this is infeasible, we will also accept a detailed set of instructions. -->

1. Generate a new empty project at https://start.spring.io/ with Language: Kotlin, Java: 11
2. Configure `gradle-docker-plugin` as simple as possible
3. Run `dockerBuildImage` task

### Stacktrace
```
> Task :dockerCreateDockerfile FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':dockerCreateDockerfile'.
> Index 0 out of bounds for length 0

* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':dockerCreateDockerfile'.
	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:38)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:63)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:49)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:46)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:416)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:406)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:102)
	at org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:46)
	at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:43)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:355)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:343)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:336)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:322)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker$1.execute(DefaultPlanExecutor.java:134)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker$1.execute(DefaultPlanExecutor.java:129)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:202)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:193)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:129)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0
	at com.bmuschko.gradle.docker.shaded.org.objectweb.asm.ClassReader.readStringish(ClassReader.java:3481)
	at com.bmuschko.gradle.docker.shaded.org.objectweb.asm.ClassReader.readClass(ClassReader.java:3495)
	at com.bmuschko.gradle.docker.shaded.org.objectweb.asm.ClassReader.accept(ClassReader.java:421)
	at com.bmuschko.gradle.docker.shaded.org.objectweb.asm.ClassReader.accept(ClassReader.java:391)
	at com.bmuschko.gradle.docker.utils.MainClassFinder.createClassDescriptor(MainClassFinder.groovy:159)
	at com.bmuschko.gradle.docker.utils.MainClassFinder.doWithMainClasses(MainClassFinder.groovy:128)
	at com.bmuschko.gradle.docker.utils.MainClassFinder.findSingleMainClass(MainClassFinder.groovy:23)
	at com.bmuschko.gradle.docker.DockerSpringBootApplicationPlugin.getSpringApplicationMainClassName(DockerSpringBootApplicationPlugin.groovy:176)
	at com.bmuschko.gradle.docker.DockerSpringBootApplicationPlugin.access$0(DockerSpringBootApplicationPlugin.groovy)
	at com.bmuschko.gradle.docker.DockerSpringBootApplicationPlugin$5$9.call(DockerSpringBootApplicationPlugin.groovy:119)
	at com.bmuschko.gradle.docker.DockerSpringBootApplicationPlugin$5$9.call(DockerSpringBootApplicationPlugin.groovy)
	at org.gradle.api.internal.provider.DefaultProvider.getOrNull(DefaultProvider.java:41)
	at com.bmuschko.gradle.docker.tasks.image.Dockerfile$StringArrayInstruction.getText(Dockerfile.groovy:927)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at org.gradle.api.internal.tasks.properties.bean.AbstractNestedRuntimeBeanNode$DefaultPropertyValue$1$1.create(AbstractNestedRuntimeBeanNode.java:81)
	at org.gradle.util.SingleMessageLogger.whileDisabled(SingleMessageLogger.java:341)
	at org.gradle.api.internal.tasks.properties.bean.AbstractNestedRuntimeBeanNode$DefaultPropertyValue$1.get(AbstractNestedRuntimeBeanNode.java:77)
	at com.google.common.base.Suppliers$NonSerializableMemoizingSupplier.get(Suppliers.java:167)
	at org.gradle.api.internal.tasks.properties.bean.AbstractNestedRuntimeBeanNode$DefaultPropertyValue.getValue(AbstractNestedRuntimeBeanNode.java:147)
	at org.gradle.api.internal.tasks.properties.bean.AbstractNestedRuntimeBeanNode$DefaultPropertyValue.validate(AbstractNestedRuntimeBeanNode.java:165)
	at org.gradle.api.internal.tasks.DefaultTaskInputPropertySpec.validate(DefaultTaskInputPropertySpec.java:68)
	at org.gradle.api.internal.tasks.execution.DefaultTaskProperties.validate(DefaultTaskProperties.java:197)
	at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:50)
	at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:109)
	at org.gradle.api.internal.tasks.execution.ResolveBeforeExecutionOutputsTaskExecuter.execute(ResolveBeforeExecutionOutputsTaskExecuter.java:67)
	at org.gradle.api.internal.tasks.execution.ResolveAfterPreviousExecutionStateTaskExecuter.execute(ResolveAfterPreviousExecutionStateTaskExecuter.java:46)
	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:93)
	at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:45)
	at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:94)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
	... 24 more
```

### Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->

Java: 11

Full Configuration I'm using:
```
plugins {
    id 'org.springframework.boot' version '2.1.3.RELEASE'
    id 'org.jetbrains.kotlin.jvm' version '1.3.21'
    id 'org.jetbrains.kotlin.plugin.spring' version '1.3.21'
    id 'com.bmuschko.docker-spring-boot-application' version '4.6.1'
}

apply plugin: 'io.spring.dependency-management'

group = 'com.example'
version = '0.0.1-SNAPSHOT'
sourceCompatibility = '1.8'

repositories {
    mavenCentral()
}

dependencies {
    implementation 'org.springframework.boot:spring-boot-starter'
    implementation 'org.jetbrains.kotlin:kotlin-reflect'
    implementation 'org.jetbrains.kotlin:kotlin-stdlib-jdk8'
    testImplementation 'org.springframework.boot:spring-boot-starter-test'
}

compileKotlin {
    kotlinOptions {
        freeCompilerArgs = ['-Xjsr305=strict']
        jvmTarget = '1.8'
    }
}

compileTestKotlin {
    kotlinOptions {
        freeCompilerArgs = ['-Xjsr305=strict']
        jvmTarget = '1.8'
    }
}

docker {
    springBootApplication {
        baseImage = 'openjdk:8-alpine'
        ports = [8080]
        tag = 'awesome-spring-boot:1.115'
    }
}
```

Main Class:
```kotlin
package com.example.demo

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication

@SpringBootApplication
class DemoApplication

fun main(args: Array<String>) {
    runApplication<DemoApplication>(*args)
}
```

--- a/src/main/groovy/com/bmuschko/gradle/docker/utils/MainClassFinder.groovy
+++ b/src/main/groovy/com/bmuschko/gradle/docker/utils/MainClassFinder.groovy
@@ -123,7 +123,7 @@ class MainClassFinder {
         stack.push(rootFolder)
         while (!stack.isEmpty()) {
             File file = stack.pop()
-            if (file.isFile()) {
+            if (file.isFile() && file.name.endsWith('.class')) {
                 InputStream inputStream = new FileInputStream(file)
                 ClassDescriptor classDescriptor = createClassDescriptor(inputStream)
                 if (classDescriptor != null && classDescriptor.isMainMethodFound()) {

--- a/src/main/groovy/com/bmuschko/gradle/docker/utils/MainClassFinder.groovy
+++ b/src/main/groovy/com/bmuschko/gradle/docker/utils/MainClassFinder.groovy
@@ -138,13 +138,13 @@ class MainClassFinder {
                 pushAllSorted(stack, file.listFiles(new FileFilter() {
                     @Override
                     boolean accept(File pathname) {
-                        file.isDirectory() && !file.getName().startsWith(".")
+                        pathname.isDirectory() && !pathname.getName().startsWith(".")
                     }
                 }))
                 pushAllSorted(stack, file.listFiles(new FileFilter() {
                     @Override
                     boolean accept(File pathname) {
-                        file.isFile() && file.getName().endsWith(DOT_CLASS)
+                        pathname.isFile() && pathname.getName().endsWith(DOT_CLASS)
                     }
                 }))
             }

--- a/src/test/groovy/com/bmuschko/gradle/docker/utils/MainClassFinderTest.groovy
+++ b/src/test/groovy/com/bmuschko/gradle/docker/utils/MainClassFinderTest.groovy
@@ -56,4 +56,16 @@ class MainClassFinderTest extends Specification {
         then:
         !mainClass
     }
+
+    def "only consider .class files"() {
+        given:
+        testJarFile.addClass('a/b/c/E.class', AnnotatedClassWithMainMethod)
+        new File(testJarFile.jarSource, 'noClass.txt').createNewFile()
+
+        when:
+        MainClassFinder.findSingleMainClass(testJarFile.getJarSource(), ANNOTATION_CLASS_NAME)
+
+        then:
+        noExceptionThrown()
+    }
 }

--- a/src/test/groovy/com/bmuschko/gradle/docker/utils/MainClassFinderTest.groovy
+++ b/src/test/groovy/com/bmuschko/gradle/docker/utils/MainClassFinderTest.groovy
@@ -63,9 +63,9 @@ class MainClassFinderTest extends Specification {
         new File(testJarFile.jarSource, 'noClass.txt').createNewFile()
 
         when:
-        MainClassFinder.findSingleMainClass(testJarFile.getJarSource(), ANNOTATION_CLASS_NAME)
+        String mainClass = MainClassFinder.findSingleMainClass(testJarFile.getJarSource(), ANNOTATION_CLASS_NAME)
 
         then:
-        noExceptionThrown()
+        mainClass == 'a.b.c.E'
     }
 }

